pkg/libvirt/resolve_vm.go

```diff
@@ -27,17 +27,11 @@ const (
 	procPath    string = "/proc/%s/task"
 )
 
-func getThreadIDsForPID(pid, extraPath string) []string {
+func getThreadIDsForPID(pid, fullPath string) []string {
```

Comment:
I see that the previous use of extraPath was more like a `basePath` (than an extraPath) which gets added in front of `procPath` - `/proc/%s/task` giving us `/extra/path/proc/%s/task` where `%s` is  replaced by `pid`.

This change requires caller to know the format string (which looks odd to me).  I also see that no other code change is make (e.g. the actual call to the getThreadIDsForPID is not modified) which makes me wonder how the fix is supposed to work 🤔

Reply:
Hi, @sthaha 

Please see the call chain. In existing code, the extraPath and procPath have same value, and it will cause `fullpath` join error.
`extraPath=/proc/%s/task`
`procPath=/proc/%s/task`
`fullPath=/proc/%s/task/proc/%s/task`

For detailed information, please see the issue https://github.com/sustainable-computing-io/kepler/issues/1008

https://github.com/sustainable-computing-io/kepler/blob/63fb1c4434cb693642fb7b024900dcdfda101a6a/pkg/collector/container_hc_collector.go#L115

https://github.com/sustainable-computing-io/kepler/blob/63fb1c4434cb693642fb7b024900dcdfda101a6a/pkg/libvirt/resolve_vm.go#L57

https://github.com/sustainable-computing-io/kepler/blob/63fb1c4434cb693642fb7b024900dcdfda101a6a/pkg/libvirt/resolve_vm.go#L81

Reply:
Hi @LeiZhou-97 
Sorry for the delay. So I checked your modification and I'm astonished the mistake passed under my radar. 
I've tested this morning and with or without the modification I've got the same result. Which is weird but explain why I did not see it. 
To be honest your modification looks better. I do not remember why I wanted to Join two paths but in the end, reading /proc/PID/task/ is enough to get the list of TID attached to the PID. 
So thank you for correction. 
Anthony

Reply:
Thanks @LeiZhou-97  that makes sense now

Reply:
@sthaha 
From my side, they have the same effect. But I don't know if Anthony may use this `procPath` const elsewhere in the future, so I keep it. And it seems will not cause performance issue.

Reply:
@sthaha @LeiZhou-97 That would be indeed the same. 
Let's wait on the re-factorization of the code and we will see if this path is used somewhere else or not.